airflow/bin/cli.py

```diff
@@ -686,10 +687,24 @@ def test(args, dag=None):
         task.params.update(passed_in_params)
     ti = TaskInstance(task, args.execution_date)
 
-    if args.dry_run:
-        ti.dry_run()
-    else:
-        ti.run(ignore_task_deps=True, ignore_ti_state=True, test_mode=True)
+    try:
+        if args.dry_run:
+            ti.dry_run()
+        else:
+            ti.run(ignore_task_deps=True, ignore_ti_state=True, test_mode=True)
+    except Exception:
+        if args.post_mortem:
+            debugger = get_debugger()
+            debugger.post_mortem()
+        else:
+            raise
+
+
+def get_debugger():
+    try:
+        return importlib.import_module("ipdb")
+    except ImportError:
+        return importlib.import_module("pdb")
```

Comment:
Do we need to have a separate function here?

I understand this is very "personal preference": but I think we can simply put this part after `if args.post_mortem:` in the `try-except` block above, given the logic inside this `get_debugger()` function is too simple.

Reply:
Agreed, I factored this back into the `test` function.